packages/pwa-buildpack/src/WebpackTools/PWADevServer.js

```diff
@@ -75,84 +73,97 @@ const PWADevServer = {
             );
         });
     },
-    async findFreeHostname(identifier, times = 0) {
-        const maybeHostname =
-            identifier + (times ? times : '') + '.local.pwadev';
-        // if it has a port, it exists
-        const exists = await PWADevServer.portsByHostname.get(maybeHostname);
-        if (!exists) {
-            debug(
-                `findFreeHostname: ${maybeHostname} unbound to port and available`
-            );
-            return maybeHostname;
+    getUniqueSubdomain(customName) {
+        let name = PWADevServer.DEFAULT_NAME;
+        if (typeof customName === 'string') {
+            name = customName;
         } else {
-            debug(`findFreeHostname: ${maybeHostname} bound to port`, exists);
-            if (times > 9) {
-                throw Error(
+            const pkgLoc = join(process.cwd(), 'package.json');
+            try {
+                // eslint-disable-next-line node/no-missing-require
+                const pkg = require(pkgLoc);
+                if (!pkg.name) {
+                    throw new Error(
+                        `package.json has ${pkg.name} "name" field!`
```

Comment:
Could we rewrite this error to be a little more clear?

Reply:
The way I've structured this, that error gets immediately caught, and then handled along with any other errors that might occur trying to `require()` the package.json file. So the errors that throw will look like:

Error loading package.json:
```
getUniqueSubdomain: Using default "my-pwa" prefix. Could not autodetect theme name from package.json Error: cannot find module './package.json'`
```

package.json exists but somehow has no "name" field:
```
`getUniqueSubdomain: Using default "my-pwa" prefix. Could not autodetect theme name from package.json Error: package.json has undefined "name" field!`
```

I can add a comment explaining this if you like, but I think the errors that emit are clear enough.

Reply:
No, my concern was that if `pkg.name` is falsy but not `undefined`, the sentence won't make much sense, since this assumes it's an adjective. 😅 

If `undefined` is the only possible falsy value for `pkg.name`, then this doesn't need to be a template string. Just write `has undefined "name" field`. Otherwise, let's make it a bit more flexible.

```
// pkg.name: ''
package.json has "name" field!

// alternative
package.json is missing a "name" value!
```

Reply:
I had not considered the null string. Changing!